### DISCUSSION TITLE
Firebase onAuthStateChanged user returns null in next build

### DISCUSSION THREAD

**Discussion by stormfar** [vercel/next.js#11333] [Help]
https://github.com/zeit/next.js/blob/cadc950e2d2eea969ddabf135e9223b2c21108d6/examples/with-firebase/context/userContext.js#L12

Following the with-firebase example, userContext.js inits firebase and returns the user as expected when running in development mode with `next`, but when serving the output of `next build`, firebase inits, but user returns `null`. I'm using SSG for most pages, so not sure if that has some effect. 

<img width="198" alt="Image 2020-03-24 at 8 04 52 PM" src="https://user-images.githubusercontent.com/4154407/77466565-c0b46e80-6e0a-11ea-9f58-aa08528e35d6.png">

**Comment by m1m6** [comment]
Just faced same issue.

**Comment by charlie632** [comment]
Following because I'm having the same issue
